EIPS/eip-rich-site-proposed-contract-metadata.md

```diff
@@ -0,0 +1,100 @@
+
+## Motivation
+
+We've got a lot of ways we try to gather the information for confirmations, but it ranges between very complicated, unreliable, centralized, and incomplete, and is not getting the job done.
+1. Contract names
+  - Local address book
+    - Requires user interaction, which adds UX burden, and most users skip.
+  - Centralized registries
+    - Are perpetually incomplete, and favor the established, not the upstart.
+   - reverse-resolved ENS names
+     - Attackers can choose familiar-looking names, so these are not a reliable basis of secure review of what you're looking at unless you entered it yourself.
+2. Method information
+  - Registry lookups (4byte directory, the Parity on-chain registry)
+    - Can be prone to collisions (like At Inversebrah, with trolling increasing)
+    - Returns ABIs
+      - Has no parameter names, leaving most of the transaction illegible.
+  - Trusting the contract for its own metadata (EIP-719, EIP-4430)
+    - Requires an active connection to the blockchain.
+    - Is not widespread (no backwards compatibility)
+  - [Transaction Insights](https://metamask.zendesk.com/hc/en-us/articles/4412543412123-Transaction-Insight)
+    - Requires an additional fairly heavy-weight process
+    - Is largely centralized today
+    - Only works for verified contracts on etherscan (although designed for eventual Sourcify support).
+
+Additionally, this method serves to help populate the wallet with trustworthy information about what it holds. It reduces the need to rely on any central services for indexing (enhancing performance and privacy), and eliminates dangers that come from index-based automatic asset detection, like receiving harrassment, or [Airdrop scams](https://medium.com/metamask/phisher-watch-airdrop-scams-82eea95d9b2a).
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Wallets may integrate the ability to propose names for contracts, methods, and parameters into any method that touches an address, so the wallet can be expanded with this metadata at runtime. These methods include `eth_sendTransaction`, and `eth_signTypedData` (and its variants).
+
+Benefits
+  - No need for any central server
+  - No registry with collisions
+  - Works on all chains
+  - Includes parameter names as well as method and contract names.
+  - Could be easily added to any existing dapp
+  - If every dapp a user touches uses this, then no phishing site could get the user to give up one of those assets without seeing a legible confirmation.
+  - Can even be implemented on a cold/offline or hardware wallet.
```

Comment:
This content sounds like motivation, not specification, and should be moved to the appropriate section.

Reply:
Done in latest commit.